moveit_core/robot_state/include/moveit/robot_state/robot_state.h

```diff
@@ -1311,27 +1298,39 @@ as the new values that correspond to the group */
   /** \brief Update the state after setting a particular link to the input global transform pose.*/
   void updateStateWithLinkAt(const LinkModel* link, const Eigen::Isometry3d& transform, bool backward = false);
 
-  const Eigen::Isometry3d& getGlobalLinkTransform(const std::string& link_name)
-  {
-    return getGlobalLinkTransform(robot_model_->getLinkModel(link_name));
```

Comment:
Why did you removed the inlining for these 1-liner functions? This (slightly) degrades performance.

Reply:
The optimizer decided what gets inlined, modern compilers do not care if its in the header or not (unless its a template)

Reply:
``inline`` isn't needed for modern compilers, either. they will choose on their own what to inline or not.

Reply:
The compiler cannot inline this function if it's not declared in the header. How should another compilation unit know what to inline? See http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines.html#Rf-inline

Reply:
i've added inline to header functions.